services/m3dbnode/config/config.go

```diff
@@ -93,48 +92,13 @@ type Configuration struct {
 	// The pooling policy.
 	PoolingPolicy PoolingPolicy `yaml:"poolingPolicy"`
 
-	// The dynamic etcd configuration for config service client.
-	ConfigService *etcdclient.Configuration `yaml:"configService"`
-
-	// The static configuration for config service client.
-	StaticConfig *StaticConfiguration `yaml:"static"`
+	// The environment (static or dynamic) configuration.
+	EnvironmentConfig *environment.Configuration `yaml:"config"`
```

Comment:
You could make this non pointer type to avoid having to check for nil-ability of this.